src/Analysis/Ast/Impl/Extensions/PythonFunctionExtensions.cs

```diff
@@ -34,5 +38,44 @@ public static bool HasClassFirstArgument(this IPythonClassMember m)
             IScope gs = f.DeclaringModule.GlobalScope;
             return gs?.TraverseBreadthFirst(s => s.Children).FirstOrDefault(s => s.Node == f.FunctionDefinition);
         }
+
+        /// <summary>
+        /// Reports any decorator errors on function.
+        /// Returns true if the decorator combinations for the property are valid
+        /// </summary>
+        public static bool HasValidDecorators(this IPythonFunctionType f, IExpressionEvaluator eval) {
+            bool valid = true;
+            // If function is abstract, allow all decorators because it will be overridden
+            if (f.IsAbstract) {
```

Comment:
I looked at [this](https://stackoverflow.com/questions/45248243/most-pythonic-way-to-declare-an-abstract-class-property) for the abstract check